### DISCUSSION TITLE
Only support 'supported' versions of sonar?

### DISCUSSION THREAD

**Discussion by hazendaz** [spotbugs/sonar-findbugs#932] [General]
Today, this project is running tests against sonar 7.9, 8.9, 9.7, 9.9, and 10.3.  Some PRs have been blocked to continue to allow obsolete software.

See https://www.sonarsource.com/products/sonarqube/downloads/lts/ or https://endoflife.date/sonar as well as look at our renovate dashboard to see what has been skipped.

If sonar doesn't support the versions anymore, I don't see why we continue to support either.  Sonar is suggesting a number of warnings and has been for some time.  For example, it has been stating that java 11 would be entirely dropped.  While its not completely yet, they did flip a flag that started breaking our builds and our code was already compatible with java 17 in building and its maven so maven controls the targeting (in fact java 21 will control back to java 8 for that matter).  To me this is the equivalent of trying to support java 7 for example when even no vendor does.

So I'm suggesting we start dropping legacy support items which should ease support.  We should at a minimum drop sonar 7.

**Comment by gtoison** [comment]
Here's some background on how the project got where it is now:
- The project is compiled targetting various versions of SonarQube, each version has its own API version, for the plugins and for the java plugin.
- The Maven pom refers to the earliest version of these API but it is overriden by the GitHub matrix build.
- Historically the API have been quite stable, however some features are marked as deprecated even though the suggested replacement does not exist (or work) yet on the latest versions
- From what I've seen the practice was to drop support for older versions when there's a breaking change, but there wasn't any for a while, so it just continues to work on 7.9
- Upgrading to more recent versions of the API in the pom would be confusing because they might not be backward compatible with SonarQube versions we want to support (9.9 LTS)
- AFAIK there's no functionnality that we're missing (or problem caused) due to remaining on older versions in the pom. Sonarqube does not use the pom to load plugins.
- I think we want to know when something breaks support for 7.9 or 8.9, and then release a new version clearly stating that older versions aren't supported anymore. The integration tests are the smoke test for that. Otherwise we have a grey area where we're not sure if it works or not
- Many orgs are running old SonarQube versions and, at the same time, have different teams (developpers, devops, admins, etc.) so it is frequently complicated to even know the versions (server or plugin) they're running. Overall it saves time (for me at least) to make sure that the plugin works reasonably well on older versions.

So to sum up, the maintained versions are the current LTS and the latest. When some breaking changes will break older versions they will be dropped

**Comment by hazendaz** [comment]
I believe this issue has now been addressed, closing.
